ocp_resources/utils.py

```diff
@@ -17,15 +17,32 @@ def __str__(self):
 
 class TimeoutSampler:
     """
-    Samples the function output.
-
-    This is a generator object that at first yields the output of function
-    `func`. After the yield, it either raises instance of `TimeoutExpiredError` or
-    sleeps `sleep` seconds.
-
-    Yielding the output allows you to handle every value as you wish.
-
-    Feel free to set the instance variables.
+    Yields a given function's return value.
+
+    Before each following iteration (the next yield) over the TimeSampler object, the generator sleeps until it reaches
+    timeout and throws a TimeoutExpiredError exception.
+
+    Args:
+        wait_timeout (int)
+        sleep (int)
+        func (callable): the function to be invoked upon each yield
+        exceptions (a single BaseException or derived classes or a tuple with multiple exceptions): expected exception
+            to ignore when running the generator code;
+            Default: Exception.
+        exceptions_msg: expected exception(s) message; if an exception is caught, but exceptions_msg could not be found
+            in the exception's expression (str), the exception (that was caught) is re-raised after recording the error.
```

Comment:
`the exception (that was caught) is re-raised after recording the error.`

I don't understand this sentence, can you please explain?

Reply:
simply explaining what happens here:
```        if self.exceptions_msg:
            if self.exceptions_msg not in str(exp):
                LOGGER.error(log)
                raise exp
            else:
                LOGGER.warning(f"{self.exceptions_msg}: Retrying")
```